app/src/main/java/com/pr0gramm/app/ui/fragments/BenisGraphFragment.kt

```diff
@@ -92,34 +93,37 @@ class BenisGraphFragment : BaseFragment("BenisGraphFragment") {
         private var maxY: Int = 0
 
         init {
-            userService.loadBenis().subscribe{
-                data.add(it)
-                minX = if(minX > it.time) it.time else minX
-                maxX = if(maxX < it.time) it.time else maxX
-                minY = if(minY > it.benis) it.benis else minY
-                maxY = if(maxY < it.benis) it.benis else maxY
-                var date = dateFormat.format(Date(it.time))
-                firstOfDays[date] = Math.min(it.time, firstOfDays.getOrDefault(date,Long.MAX_VALUE))
-
-                val deltaX = (maxX - minX)*0.05
-
-                plot.outerLimits.set(
-                        minX,
-                        maxX,
-                        minY,
-                        maxY*1.2
-                )
-                plot.setRangeBoundaries(
-                        minY,
-                        maxY*1.2,
-                        BoundaryMode.FIXED
-                )
-                plot.setDomainBoundaries(
-                        minX,
-                        maxX,
-                        BoundaryMode.FIXED
-                )
-                plot.redraw()
+            Observable.fromCallable{ userService.loadBenis()}
+                    .subscribeOn(BackgroundScheduler.instance())
+                    .subscribe{
+                for (br in it){
```

Comment:
bin jetzt soweit gekommen. habe auch mal in BenisRecord ne ThreadSleep gepackt (sieht man ja - ich Depp habs commited).